### DISCUSSION TITLE
How to reuse vpp-agent as a non-vpp container agent?

### DISCUSSION THREAD

**Discussion by OlympusHo** [ligato/vpp-agent#953] [Q&A]
I want to develop a agent for non-vpp container,how to reuse vpp-agent?

**Comment by ondrej-fabry** [comment]
Could you clarify? Do you want to develop an app based on vpp-agent or do you want to configure vpp-agent for non-vpp container?

**Comment by OlympusHo** [comment]
> Could you clarify

I want to configure vpp-agent for non-vpp container

**Comment by ondrej-fabry** [comment]
Then this previously asked question might be useful for you: https://github.com/ligato/vpp-agent/issues/796

**Comment by OlympusHo** [comment]
#796 has mention the example of https://github.com/ligato/vppagent/blob/master/examples/localclient_linux/veth/main.go,but it still include the vpp plugin and just suitable for local scenes ,not watch the etcd.I think that I write a agent just include linuxPlugin and linuxPlugin.deps.Watcher = kvdbsync.NewPlugin(kvdbsync.UseKV(&etcd.DefaultPlugin)) can achieve my question，do you think it's ok?

**Comment by ondrej-fabry** [comment]
Well, it depends on what you want to achieve. 

1. You want to manage some non-VPP container (but have some VPP container as well)
  - this is similar case to the example posted above
2. You want to manage only non-VPP containers (no VPP container at all)
  - at the moment, I'm afraid you are don't have any option here when running stock image
  - however, you could build your own agent, in which you would disable VPP plugin inside [app](https://github.com/ligato/vpp-agent/blob/master/app/vpp_agent.go), although this is not tested and has not been used like this before, AFAIK

Perhaps, you could describe your goal , so I could better understand your use-case.

**Comment by OlympusHo** [comment]
Thank you very much for your reply!
My use-case is "manage some non-VPP container (but have some VPP container as well)",
Not only manage some interfaces in non-vpp container ,but also manage my app(e.g. dpi,dnsmasq) in non-vpp container,so i want to build my own nonvpp-agent.

**Comment by ondrej-fabry** [comment]
So if I understand correctly. You want to manage some services in linux (container), which is some other container and not the one with VPP/vpp-agent running, right?

At the moment for Linux configuration we support:
- interfaces (veth, tap)
- static routes
- static arps

All of these can be also configured for other containers by defining `namespace` field. When using `NamespaceType=MICROSERVICE_REF_NS` and setting microservice=xxx, other containers are automatically watched using docker client and when container with matching `MICROSERVICE_LABEL=xxx` appears, all configuration for this container is applied to its namespace by looking up its PID linux network namespace.

From you uses case I understood that you want to manage some services running in other containers?
